docs/data/joy/components/input/input.md

```diff
@@ -62,6 +62,42 @@ You can add standard form attributes such as `required` and `disabled` to the In
 
 {{"demo": "InputFormProps.js"}}
 
+### Focused ring
+
+Provide these CSS variables to `sx` prop to control the focused ring appearance:
+
+- `--Input-focusedInset`: the focused ring's **position**, either inside(`inset`) or outside(`var(--any, )`) of the Input.
+- `--Input-focusedThickness`: the **size** of the focused ring.
+- `--Input-focusedHighlight`: the **color** of the focused ring.
+
+{{"demo": "FocusedRingInput.js"}}
+
+:::success
+To get full control of the focused ring, customize the `box-shadow` of the `::before` pseudo element directly
+
+```js
+<Input sx={{ '&:focus-within(::before)': { boxShadow: '...your custom value' } }} />
+```
+
+:::
+
+#### Debugging the focus ring
+
+To display the Input's focus ring by simulating user's focus, inspect the Input element and toggle the [pseudostate panel](https://developer.chrome.com/docs/devtools/css/#pseudostates).
+
+- If you inspect the Input's root element, with `.MuiInput-root` class, you have to toggle on the `:focus-within` state.
+- If you inspect the `<input>` element, you have to toggle on the `:focus` state.
+
+### Triggering the focus ring
+
+To trigger the focus ring programmatically, set the CSS variable `--Input-focused: 1`.
+
+{{"demo": "TriggerFocusInput.js"}}
+
+:::info
+💡 The focus ring still appear on focus even though you set `--Input-focused: 0`.
```

Comment:
Regarding this statement, I checked that if we set `--Input-focused: 0`, the focused is removed. Is this correct? The thickness is multiplied by zero.

Reply:
Yes, but you will see the focus ring if you focus on the input because `:focus-within` has higher CSS specificity.

https://github.com/mui/material-ui/assets/18292247/2886beb9-437a-46f1-b7ed-27c4b337a317